### ISSUE TITLE
pca with varimax rotation

### ISSUE THREAD

**Issue by matarhaller** [juanshishido/okcupid#5]
doesn't seem to be implemented in python.

**Comment by matarhaller** [comment]
I'm trying to implement R in python using rpy2, but there is also this which seems to maybe be an implementation of varimax rotation
https://github.com/rossfadely/consomme/blob/master/consomme/rotate_factor.py

**Comment by matarhaller** [comment]
running it using rpy2 is super duper slow.
going to try this:
http://stackoverflow.com/questions/17628589/perform-varimax-rotation-in-python-using-numpy

**Comment by juanshishido** [comment]
Did `rotate_factor.py` not work well?

Sweet re the SO implementation using NumPy. Good luck!

**Comment by juanshishido** [comment]
Just wanted to mention that we could also consider nonlinear dimensionality reduction techniques, such as Kernel PCA. I don't know whether our features are linearly separable, so thought I'd bring this up.

**Comment by matarhaller** [comment]
Good point. I was thinking about that, but thought we'd start simple for
now.

**Comment by juanshishido** [comment]
I read some more after posting that and found out that text is usually linearly separable, which is why I deleted my comment :sweat_smile:

**Comment by matarhaller** [comment]
`rotate_factor.py` only works on a square matrix.

**Comment by jnaras** [comment]
Mmmm. Can we try it in R?

**Comment by matarhaller** [comment]
I tried it and it seemed like it didn't like the zeros in the matrix. I'm
not familiar enough with rotation to know if that's the true issue though.

**Comment by jnaras** [comment]
Okay, I'll try to write it out in matrix form then and plug it into your PCA code!
